store/tikv/client.go

```diff
@@ -212,10 +214,19 @@ type rpcClient struct {
 	conns    map[string]*connArray
 	security config.Security
 
-	idleNotify uint32
-	// Periodically check whether there is any connection that is idle and then close and remove these idle connections.
+	dieNotify uint32
+	// Periodically check whether there is any connection that is die and then close and remove these connections.
 	// Implement background cleanup.
-	isClosed bool
+	isClosed    bool
+	dieListener func(addr []string)
+}
+
+var _ ListenableRPCClient = &rpcClient{}
+
+// ListenableRPCClient present a RPCClient can be listen.
+type ListenableRPCClient interface {
```

Comment:
I think we don't need to add this abstraction, we can check if it is `*rpcClient` directly.